lib/publiccloud/ssh_interactive.pm

```diff
@@ -144,6 +144,7 @@ sub prepare_ssh_tunnel {
 
     # Permit root passwordless login over SSH
     $instance->ssh_assert_script_run('sudo sed -i "s/PermitRootLogin no/PermitRootLogin prohibit-password/g" /etc/ssh/sshd_config');
+    $instance->ssh_assert_script_run('sudo sed -iE "/^AllowTcpForwarding/c\AllowTcpForwarding yes" /etc/ssh/sshd_config') if (get_var('FLAVOR') =~ 'EC2' && get_var('FLAVOR') =~ 'Hardened');
```

Comment:
```suggestion
    $instance->ssh_assert_script_run('sudo sed -iE "/^AllowTcpForwarding/c\AllowTcpForwarding yes" /etc/ssh/sshd_config') if (is_ec2() && is_hardended());
```

Reply:
We don't have `is_hardended()` anymore after Pavel encountered some nasty Perl errors.

Reply:
ok

Reply:
let's just merge it :)

Reply:
No. It should be done for all hardened images, not just for EC2.
There is also better solution in #16160